app/assets/javascripts/discourse/app/components/modal/upsert-hyperlink.gjs

```diff
@@ -145,7 +145,15 @@ export default class UpsertHyperlink extends Component {
   @action
   onFormSubmit(data) {
     const origLink = data.linkUrl;
-    const linkUrl = encodeURI(prefixProtocol(origLink));
+    let linkUrl = prefixProtocol(origLink);
+
+    if (typeof linkUrl === "string") {
+      try {
+        linkUrl = encodeURI(decodeURI(linkUrl));
+      } catch {
+        linkUrl = encodeURI(linkUrl.replace(/%(?![0-9A-Fa-f]{2})/g, "%25"));
```

Comment:
I think we should also add test coverage for this too to ensure that it is working.